lib/chef/provider/dsc_resource.rb

```diff
@@ -155,6 +158,31 @@ def invoke_resource(method, output_format=:object)
         cmdlet.run!
       end
 
+      def return_dsc_resource_result(result, property_name)
+        if result.return_value.is_a?(Array)
+          # WMF Feb 2015 Preview
+          result.return_value[0][property_name]
+        else
+          # WMF April 2015 Preview
+          result.return_value[property_name]
+        end
+      end
+
+      def notify_reboot
+        @reboot_resource = Chef::Resource::Reboot.new(
+          "Reboot for #{@new_resource.name}", 
+          run_context
+        ).tap do |r|
+          r.reason("Reboot for #{@new_resource.resource}.")
+        end
+      end
+
+      def invoke_reboot
+        unless @reboot_resource.nil?
+          Chef::Log.debug("Requesting node reboot with #{@new_resource.allow_reboot}")
```

Comment:
We may not actually request a reboot a reboot here, depending on `@new_resource.allow_reboot`. Can you put two log messages here, one for if we're going to actually trigger the reboot, and another for if one was requested but we're ignoring it due to the value of allow_reboot?